btservice/AndroidBluetoothServer.cpp

```diff
@@ -103,115 +68,147 @@ void AndroidBluetoothServer::writeSocketInfoMessage()
         }
     }
 
-    btservice::proto::SocketInfo socketInfo;
-    socketInfo.set_address(ipAddr.toStdString());
-    socketInfo.set_port(5000);
-    socketInfo.set_unknown_1(0);
+    btservice::proto::SocketInfoRequest socketInfoRequest;
+    socketInfoRequest.set_ip_address(ipAddr.toStdString());
+    OPENAUTO_LOG(info) << "[AndroidBluetoothServer] ipAddress: "<< ipAddr.toStdString();
+
+    socketInfoRequest.set_port(5000);
+    OPENAUTO_LOG(info) << "[AndroidBluetoothServer] port: "<< 5000;
 
-    if(this->writeProtoMessage(7, socketInfo))
+    if(this->writeProtoMessage(1, socketInfoRequest))
     {
-        OPENAUTO_LOG(info) << "[AndroidBluetoothServer] Sent socket info.";
-        handshakeState_ = ConnectionStatus::SENT_SOCKETINFO_MESSAGE;
+        OPENAUTO_LOG(info) << "[AndroidBluetoothServer] Sent SocketInfoRequest.";
     }
     else
     {
-        OPENAUTO_LOG(error) << "[AndroidBluetoothServer] Error sending socket Info.";
-        handshakeState_ = ConnectionStatus::ERROR;
+        OPENAUTO_LOG(error) << "[AndroidBluetoothServer] Error sending SocketInfoRequest.";
     }
 }
+void AndroidBluetoothServer::writeSocketInfoResponse()
+{
+    OPENAUTO_LOG(info) << "[AndroidBluetoothServer] Sending SocketInfoResponse.";
+    QString ipAddr;
+    foreach(QHostAddress addr, QNetworkInterface::allAddresses())
+    {
+        if(!addr.isLoopback() && (addr.protocol() == QAbstractSocket::IPv4Protocol))
+        {
+            ipAddr = addr.toString();
+        }
+    }
+
+    btservice::proto::SocketInfoResponse socketInfoResponse;
+    socketInfoResponse.set_ip_address(ipAddr.toStdString());
```

Comment:
excuse my laziness here because i dont feel like digging through the code, but is passing an empty string into `set_ip_address` safe?

Reply:
If the code reaches this point and ipAddr is empty, there's a bigger issue with the setup :)
But even then, it won't cause things to crash, the handshake would be expected to fail though.

Reply:
yeah i was just moreso wondering about a non-traditional setup (for example running in docker or something without a networking stack)

but as long as it wont crash then its good with me